streams/src/main/java/org/apache/kafka/streams/kstream/Windows.java

```diff
@@ -36,11 +36,11 @@
  * @see SessionWindows
  * @see TimestampExtractor
  */
+@SuppressWarnings("deprecation")
 public abstract class Windows<W extends Window> {
 
-    // By default grace period is 24 hours for all windows,
-    // in other words we allow out-of-order data for up to a day
-    protected static final long DEFAULT_GRACE_PERIOD_MS = 24 * 60 * 60 * 1000L;
+    // By default, there is no longer any grace period since Kafka 3.0
+    protected static final long DEFAULT_GRACE_PERIOD_MS = 0L;
```

Comment:
I think this comment might confuse other developers. Could you explain more about zero default grace period?

Reply:
I am going to address that now, with the new code version